tutorials/mpi-fortran/code/reduce_avg.f90

```diff
@@ -0,0 +1,78 @@
+module subs
+  implicit none
+contains
+  subroutine create_rand_nums(rand_nums, num_elements)
+    ! Creates an array of random numbers. Each number has a value from 0 - 1
+    integer, intent(in) :: num_elements
+    real, intent(out)   :: rand_nums(num_elements)
+
+    integer :: i
+
+    do i = 1, num_elements
+      rand_nums(i) = rand()
+    end do
+
+  end subroutine create_rand_nums
+end module subs
```

Comment:
As per previous comments, what is the benefit of this module over just inlining the necessary code where it is needed?